### PR TITLE
Add rocm build for UBI

### PR DESCRIPTION
**Pull request by Xaenalt**
Addresses: https://issues.redhat.com/browse/RHOAIENG-9003

### CODE REVIEW

File: Dockerfile.rocm [hunk 1]
```diff
@@ -1,15 +1,31 @@
...
+
+FROM ${BASE_IMAGE}
+
+RUN echo "Base image is ${BASE_IMAGE}"
+
+RUN <<EOF
+cat <<EOD > /etc/yum.repos.d/amdgpu.repo
+[amdgpu]
+name=amdgpu
+baseurl=https://repo.radeon.com/amdgpu/6.1.2/rhel/9.4/main/x86_64/
```
Comment (dtrifiro) [inline Dockerfile.rocm:11]: Do we have a way of using ubi latest instead of hardcoding `9.4` here?

File: Dockerfile.rocm [hunk 2]
```diff
@@ -27,36 +43,27 @@ ARG BUILD_FA="1"
...
-    git \
-    bzip2 \
-    libx11-6 \
-    build-essential \
-    wget \
-    unzip \
-    nvidia-cuda-toolkit \
-    tmux \
- && rm -rf /var/lib/apt/lists/*
+RUN yum -y install git bzip2 libX11-devel wget unzip https://mirror.stream.centos.org/9-stream/BaseOS/x86_64/os/Packages/tmux-3.2a-4.el9.x86_64.rpm && yum clean all
```
Comment (dtrifiro) [inline Dockerfile.rocm:49]: Do we really need `tmux`? I'd remove it. Possibly even `libX11-devel`

```suggestion
RUN yum -y install git bzip2 wget unzip
```

File: Dockerfile.rocm [hunk 3]
```diff
@@ -67,17 +74,10 @@ RUN if [ "$BUILD_FA" = "1" ]; then \
     && git checkout ${FA_BRANCH} \
     && git submodule update --init \
     && export GPU_ARCHS=${FA_GFX_ARCHS} \
-    && if [ "$BASE_IMAGE" = "rocm/pytorch:rocm5.7_ubuntu22.04_py3.10_pytorch_2.0.1" ]; then \
```
Comment (dtrifiro) [inline Dockerfile.rocm:70]: Any idea to what this patch was for?

File: Dockerfile.rocm [hunk 4]
```diff
@@ -67,17 +74,10 @@ RUN if [ "$BUILD_FA" = "1" ]; then \
...
-        patch /opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/utils/hipify/hipify_python.py hipify_patch.patch; fi \
```
Comment (dtrifiro) [inline Dockerfile.rocm:71]: Any idea to what this patch was for?

File: Dockerfile.rocm [hunk 5]
```diff
@@ -102,13 +102,19 @@ ENV RAY_EXPERIMENTAL_NOSET_ROCR_VISIBLE_DEVICES=1
 
 ENV VLLM_NCCL_SO_PATH=/opt/rocm/lib/librccl.so
 
+RUN yum -y remove python3-requests
```
Comment (dtrifiro) [inline Dockerfile.rocm:105]: Why is this required?

File: Dockerfile.rocm [hunk 6]
```diff
@@ -102,13 +102,19 @@ ENV RAY_EXPERIMENTAL_NOSET_ROCR_VISIBLE_DEVICES=1
...
+ENV PYTORCH_ROCM_ARCH=${FA_GFX_ARCHS}
+
 RUN --mount=type=cache,target=/root/.cache/pip \
     pip install -U -r requirements-rocm.txt \
-    && patch /opt/rocm/include/hip/amd_detail/amd_hip_bf16.h ./rocm_patch/rocm_bf16.patch \
     && python3 setup.py install \
-    && cp build/lib.linux-x86_64-cpython-39/vllm/_C.abi3.so vllm/ \
-    && cp build/lib.linux-x86_64-cpython-39/vllm/_punica_C.abi3.so vllm/ \
-    && cp build/lib.linux-x86_64-cpython-39/vllm/_moe_C.abi3.so vllm/ \
+    && cp build/lib.linux-x86_64-3.9/vllm/_C.abi3.so vllm/ \
```
Comment (dtrifiro) [inline Dockerfile.rocm:115]: Might be worth it to directly `pip install .` instead of manually copying `.so`s

**Comment by dtrifiro** [comment]
Smoke tests are going to fail until a new vllm-tgis-adapter release (0.5.4 compatibility) is out

**Comment by Xaenalt** [comment]
Superceded by other PR with proper build
